### PR TITLE
Fix incorrect syntax error for 'matrix'

### PR DESCRIPTION
**Pull request by jpirnay**
Remove incorrect raising of a syntax error, fixes [#882](https://github.com/meerk40t/meerk40t/issues/822)

### CODE REVIEW

File: meerk40t/core/elements.py [hunk 1]
```diff
@@ -4102,8 +4102,8 @@ def element_matrix(
             if len(data) == 0:
                 channel(_("No selected elements."))
                 return
-            if ty:
-                raise SyntaxError
+            # if ty:
```
[the remark this answers is not in the document]
Comment (Sophist-UK) [reply meerk40t/core/elements.py:4105]: Hmm - there is some other global exception handling happening elsewhere - now you don't even get a Syntax error!!

[the remark this answers is not in the document]
Comment (Sophist-UK) [reply meerk40t/core/elements.py:4105]: Ok - I tried using the transform boxes in the Navigation window when the `except SyntaxError:` had been commented out, and I didn't get either a transform or a console error message. Maybe it was my testing at fault, but if it was giving a SyntaxError before and nothing else had changed, I would have expected an error again.

> but in reality it casts to svgelements.Matrix with
> Matrix sx ky kx sy tx ty

I have no idea whether this is a bug or not. svgelements should follow the svg 1.1 specification for matrix so perhaps you could check what that says, then check the svgelements code to see what order it is expecting, then see if this is a bug in MK passing parameters in the wrong order.

[the remark this answers is not in the document]
Comment (tatarize) [reply meerk40t/core/elements.py:4105]: This is accurate the ty should be true, not raising syntax error on true. And the matrix ordering should be sx kx ky sy tx ty.

File: meerk40t/core/elements.py [hunk 2]
```diff
@@ -4085,7 +4085,7 @@ def element_resize(command, channel, _, x_pos, y_pos, width, height, data=None,
         def element_matrix(
             command, channel, _, sx, kx, sy, ky, tx, ty, data=None, **kwargs
         ):
-            if tx is None:
+            if ty is None:
```
Comment (tatarize) [inline meerk40t/core/elements.py:4088]: ty here is the last value, and if none then all values weren't given.

File: meerk40t/core/elements.py [hunk 3]
```diff
@@ -4102,14 +4102,15 @@ def element_matrix(
             if len(data) == 0:
                 channel(_("No selected elements."))
                 return
-            if ty:
-                raise SyntaxError
```
Comment (tatarize) [inline meerk40t/core/elements.py:4106]: This syntax error might have wanted `if not ty` but it's better to just list the values when you type `matrix` without operands.

File: meerk40t/core/elements.py [hunk 4]
```diff
@@ -4102,14 +4102,15 @@ def element_matrix(
...
             try:
+                # SVG 7.15.3 defines the matrix form as:
+                # [a c  e]
+                # [b d  f]
```
Comment (tatarize) [inline meerk40t/core/elements.py:4108]: This is the correct ordering here. It's svg which does sx, kx, ky, sy, tx, ty. It's not what I was used to but this is how both wxPython matrix and svg's matrixes are supposed to work.

**Review by tatarize — DISMISSED** [review]

**Review by tatarize — APPROVED** [review]

**Pull request merged**